diff --git a/lib/crypto/utils-webcrypto.js b/lib/crypto/utils-webcrypto.js
index 0433f010c3a4042ce91dff610549f90d5f0ba2b1..0b73ff84cd2a66bd9325c1735c36f1ec427b263e 100644
--- a/lib/crypto/utils-webcrypto.js
+++ b/lib/crypto/utils-webcrypto.js
@@ -13,7 +13,9 @@ module.exports = {
  * The Web Crypto API - grabbed from the Node.js library or the global
  * @type Crypto
  */
-const webCrypto = nodeCrypto.webcrypto || globalThis.crypto
+// Source: https://github.com/brianc/node-postgres/issues/3206#issuecomment-2135313773
+// PATCH: Don't break under Next.js Edge runtime production build
+const webCrypto = ("webcrypto" in nodeCrypto ? nodeCrypto.webcrypto : undefined) ?? globalThis.crypto
 /**
  * The SubtleCrypto API for low level crypto operations.
  * @type SubtleCrypto
